backend/python/app/utilities/generate_csv.py

```diff
@@ -0,0 +1,126 @@
+""" 
+Generates a csv string given a list of dictionaries
+Some Notes: 
+1. Unwind only unwinds a single level (i.e a list)
+2. CSV requires all dictionaries in the list are of the same type
+"""
+
+import csv
+import io
+
+
+def flatten_objects(dictionary, parent_key="", sep="."):
+    """
+    Flatten a dictionary of dictionaries into a single dictionary
+    :param dictionary: dictionary to flatten
+    :param parent_key: key to use as the parent key for the new dictionary
+    :param sep: separator to use when combining keys
+    :return: flattened dictionary
+    :rtype: dict
+    """
+    items = []
+    for key, val in dictionary.items():
+        new_key = parent_key + sep + key if parent_key else key
+        if isinstance(val, dict):
+            items.extend(flatten_objects(val, new_key, sep=sep).items())
+
+        else:
+            items.append((new_key, val))
+    return dict(items)
+
+
+def flatten_lists(dictionary, sep="."):
+    """
+    Flatten a dictionary of lists into a single dictionary
+    :param dictionary: dictionary to flatten
+    :param sep: separator to use when combining keys
+    :return: flattened dictionary
+    :rtype: dict
+    """
+    items = []
+    lsts = 0
+    for key, val in dictionary.items():
+        if isinstance(val, list):
+            for i in range(len(val)):
+                lsts += 1
+                new_key = key + sep + str(i)
+                items.append((new_key, val[i]))
+        else:
+            items.append((key, val))
+
+    return dict(items) if lsts == 0 else flatten_lists(dict(items), sep=sep)
+
+
+def transform_function(dict_list, transform):
+    """
+    Applies a function to each dictionary in a list of dictionaries
+    :param dict_list: list of dictionaries
+    :param transform: transformation function
+    :return: transformed dictionary list
+    :rtype: list of dict
+    """
+    new_dict_list = [transform(dictionary) for dictionary in dict_list]
+    return new_dict_list
+
+
+def unwind_field(list_of_dict, field):
+    """
+    Unwinds lists inside dicts into multiple dictionaries, returning a new list at the end
+    :param list_of_dict: list of dictionaries
+    :param field: field to unwind
+    :return: list of dictionaries
+    :rtype: list of dict
+    """
```

Comment:
I know you have tests but would be nice to add an example in the docstrings, took me a while to understand what unwind means